toolkit/types/cvss/cvss_v2.go

```diff
@@ -0,0 +1,247 @@
+package cvss
+
+import (
+	"encoding"
+	"fmt"
+	"strings"
+)
+
+// V2 is a CVSS version 2 score.
+type V2 struct {
+	mv [numV2Metrics]byte
+}
+
+var (
+	_ encoding.TextMarshaler   = (*V2)(nil)
+	_ encoding.TextUnmarshaler = (*V2)(nil)
+	_ fmt.Stringer             = (*V2)(nil)
+)
+
+// ParseV2 parses the provided string as a v2 vector.
+func ParseV2(s string) (v V2, err error) {
+	return v, v.UnmarshalText([]byte(s))
+}
+
+// UnmarshalText implements [encoding.TextUnmarshaler].
+func (v *V2) UnmarshalText(text []byte) error {
+	disallow := func(_ string) error {
+		return fmt.Errorf("unknown metric %q", "CVSS")
+	}
+	err := parseStringLax(v.mv[:], disallow, v2Rev, string(text))
+	if err != nil {
+		return fmt.Errorf("cvss v2: %w", err)
+	}
+	for m, b := range v.mv[:V2Availability] {
+		if b == 0 {
+			return fmt.Errorf("cvss v2: %w: missing metric: %q", ErrMalformedVector, V3Metric(m).String())
+		}
+	}
+	return nil
+}
+
+// MarshalText implements [encoding.TextMarshaler].
+func (v *V2) MarshalText() (text []byte, err error) {
+	// CVSSv2 vectors are not prefixed.
+	return marshalVector[V2Metric]("", v)
+}
+
+func v2Unparse(m V2Metric, c byte) string {
+	switch m {
+	case V2Exploitability:
+		switch c {
+		case 'P':
+			return "POC"
+		case 'N':
+			return "ND"
+		}
+	case V2RemediationLevel:
+		switch c {
+		case 'O', 'T':
```

Comment:
Shouldn't this be:
```go
	case V2RemediationLevel:
		switch c {
		case 'O':
			return "OF"
		case 'T':
			return "TF"
		case 'N':
			return "ND"
		}
```
To allow proper roundtripping of vectors such as `AV:N/AC:H/Au:N/C:N/I:N/A:C/E:F/RL:OF`?

Reply:
Hm, I think so.